sample/src/sidewalk_workitems.c

```diff
@@ -7,29 +7,29 @@
 #include <sidewalk_version.h>
 #include <pal_init.h>
 
-#include <sidewalk_common.h>
+#include <main.h>
 #include <sidewalk_callbacks.h>
 #include <sidewalk_workitems.h>
 #include <queue.h>
 
+static app_ctx_t* app_ctx;
+
 #define CREATE_WORKITEM(NAME, ...) static void NAME##_handler(struct k_work* work) __VA_ARGS__ \
                                     K_WORK_DEFINE(NAME, NAME##_handler)
 
+#define ADD_WORKITEM(NAME) app_ctx->NAME = &NAME
+
 LOG_MODULE_REGISTER(work, LOG_LEVEL_DBG);
 
 CREATE_WORKITEM(sidewalk_event, {
-    sid_process(app_ctx.handle);
+    sid_process(app_ctx->handle);
```

Comment:
Use Zephyr macro CONTAINER_OF to retrieve the structure that holds the work item. This structure is app_ctx_t, so you do not need global app_ctx pointer.